### PR TITLE
✨ Is2805/can disable exporter with envs (⚠️ devops) 

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

We can now disable webserver's exporter plugin via env vars as ``WEBSERVER_EXPORTER=null``.

- @mrnicegyu11 @Surfict Please adjust configs in different deployments accordingly and set ``WEBSERVER_EXPORTER=null`` where required to disable export feature  (⚠️ devops) 
- ~~@odeimaiz since the front-end needs to disable some components as well,  should I add a flag with the to check if the exporter plugin is enabled/disabled in the static data?~~ DONE (see below)
- @GitHK @sanderegg This is also a first (simple) test  of the plugin mechanism based on the new settings that we will be using later to decouple the full exporter (and gc, etc) 

## Related issue/s

Until proper fix, it temporary avoids issues in
-  #2805 
-  ITISFoundation/osparc-issues#560

## How to test

If webserver started wiht  ``INFO`` it should log
```log
INFO: simcore_service_webserver.application_settings:setup_settings(191) - Captured app settings:
{
 "API_VERSION": "0.7.0",
 "APP_NAME": "simcore_service_webserver",
 "API_VTAG": "v0",
 
  # ...

 "WEBSERVER_EXPORTER": null,

  # ... 
```

## Checklist

- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Database migration script? ``cd packages/postgres-database``, ``make setup-commit``, ``sc-pg review -m "my changes"``
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

**Comment by KZzizzle** [comment]
thanks for the notice!

**Comment by odeimaiz** [comment]
> * @odeimaiz since the front-end needs to disable some components as well,  should I add a flag with the to check if the exporter plugin is enabled/disabled in the static data?

👌

**Comment by pcrespov** [comment]
> > * @odeimaiz since the front-end needs to disable some components as well,  should I add a flag with the to check if the exporter plugin is enabled/disabled in the static data?
> 
> 👌

I added ``pluginsDisabled`` that lists the name of plugins that are disabled in the backend.

```json
{
 "apiVersion": "0.7.0",
 "appName": "simcore_service_webserver",
 "buildDate": "2022-01-09T12:26:29Z",
 "vcsRef": "dd536f998",
 "vcsUrl": "git@github.com:ITISFoundation/osparc-simcore.git",
 "stackName": "master-simcore",
 "pluginsDisabled": [
  "WEBSERVER_EXPORTER",
  "WEBSERVER_SCICRUNCH"
 ]
}
```

File: services/web/server/src/simcore_service_webserver/application_settings.py [hunk 1]
```diff
@@ -182,6 +188,18 @@ def to_client_statics(self) -> Dict[str, Any]:
             exclude_none=True,
             by_alias=True,
         )
+
+        # gets names of disabled plugins
+        plugins_disabled = []
+        # NOTE: this list is limited for security reasons. An umbounded list
+        # might reveal critical info on the settings of a deploy.
+        selected_plugins = ["WEBSERVER_EXPORTER", "WEBSERVER_SCICRUNCH"]
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/application_settings.py:196]: Maybe move this to a constant outside the scope of the function. Also I'd name something on the line of `PLUGIN_CANDIDATES` or `DISABLE_PLUGIN_CANDIDATES`

File: services/web/server/src/simcore_service_webserver/exporter/request_handlers.py [hunk 2]
```diff
@@ -88,6 +88,10 @@ async def import_project(request: web.Request):
     # bumping this requests's max size
     # pylint: disable=protected-access
     exporter_settings = get_settings(request.app)
+    assert (  # nosec
+        exporter_settings is not None
+    ), "this call was not expected with a disabled plugin"  # nosec
+
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/exporter/request_handlers.py:94]: I would actually raise and error to block the request properly. No assertion.

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/exporter/request_handlers.py:94]: I would also raise an error for `duplicate_project` and `export_projec` in this module if settings are disabled.

File: services/web/server/tests/integration/01/test_exporter.py [hunk 3]
```diff
@@ -140,20 +143,26 @@ def client(
     postgres_with_template_db: aiopg.sa.engine.Engine,
     mock_orphaned_services: mock.Mock,
     monkeypatch_setenv_from_app_config: Callable,
+    monkeypatch,
```
Comment (GitHK) [inline services/web/server/tests/integration/01/test_exporter.py:146]: is this required here?

**Review by GitHK — CHANGES_REQUESTED** [review]
Nice!
I'd like that all requests in the exporter to fail as fast as possible if the exporter is disabled.

**Review by mrnicegyu11 — APPROVED** [review]
lgtm thanks for this

**Review by odeimaiz — APPROVED** [review]
Thanks!

**Comment by pcrespov** [comment]
> I'd like that all requests in the exporter to fail as fast as possible if the exporter is disabled.

When the plugin is disabled, the app has no routes to export projects and therefore it responds I guess with a [404 Not Found](https://httpstatuses.com/404) or [405 Method Not Allowed](https://httpstatuses.com/405)  as soon as it is requested ...

Is there any other way to use the exporter other than through the designated entrypoints?

**Comment by GitHK** [comment]
> > I'd like that all requests in the exporter to fail as fast as possible if the exporter is disabled.
> 
> When the plugin is disabled, the app has no routes to export projects and therefore it responds I guess with a [404 Not Found](https://httpstatuses.com/404) or [405 Method Not Allowed](https://httpstatuses.com/405) as soon as it is requested ...
> 
> Is there any other way to use the exporter other than through the designated entrypoints?

Yes, we also discussed about it. No there are no other ways to use it.

File: services/web/server/src/simcore_service_webserver/exporter/request_handlers.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/exporter/request_handlers.py:94]: @GitHK these are request handlers and they are **not** added to the routes when the plugin is disabled.
In any case, if the plugin is disabled and for some reason there is an uexpected call to these functions, exporter_settings is actually going to be None and nothing else will work below ... 
IMO there is no need to put more errors

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/exporter/request_handlers.py:94]: totally forgot about the routes removal.

**Review by GitHK — APPROVED** [review]
👍

**Pull request merged**